opm/simulators/flow/ReservoirCouplingMaster.hpp

```diff
@@ -53,10 +56,12 @@ class ReservoirCouplingMaster {
 
     const Parallel::Communication &comm_;
     const Schedule& schedule_;
+    std::time_t start_date_;  // Master process' simulation start date
```

Comment:
Is there a specific reason for the type or should we maybe use the same type for start date everywhere? Seems like it is  `boost::gregorian::date` is SimulationTimer.